Sources/TuistKit/Services/DocService.swift

```diff
@@ -0,0 +1,103 @@
+import Foundation
+import RxBlocking
+import Signals
+import TSCBasic
+import TuistCore
+import TuistSupport
+import TuistDoc
+
+// MARK: - DocServicing
+
+protocol DocServicing {
+    func run(path: AbsolutePath) throws
+}
+
+// MARK: - DocService
+
+struct DocService {
+    private static var temporaryDirectory: AbsolutePath?
+    
+    private let projectGenerator: ProjectGenerating
+    private let swiftDocController: SwiftDocControlling
+    private let opener: Opening
+    private let fileHandler: FileHandling
+
+    init(projectGenerator: ProjectGenerating = ProjectGenerator(),
+         swiftDocController: SwiftDocControlling = SwiftDocController(),
+         opener: Opening = Opener(),
+         fileHandler: FileHandling = FileHandler())
+    {
+        self.projectGenerator = projectGenerator
+        self.swiftDocController = swiftDocController
+        self.opener = opener
+        self.fileHandler = fileHandler
+    }
+
+    func run(path: AbsolutePath, target targetName: String) throws {
+        let (_, graph, _) = try projectGenerator.loadProject(path: path)
+        
+        guard let path = graph.targetPath(name: targetName) else {
```

Comment:
I think this should not use `targetPath` but rather leverage `target.sources` property - the reasoning is as follows: If I run `tuist doc` for my project, the documentation gets generated for eg `Carthage` folder, too, since it is in the same directory as my Sources. To expand on this example, my project looks like this:
```
Project.swift
MyProjectSources/
Carthage/
``` 

In my `Project.swift` I define that all my sources lie in `MyProjectSources/**`. That's why I'd expect that the documentation is only generated for them. 

I also wonder if running `tuist doc` without passing argument should not work, too - could we maybe generate documentation for all the targets? If this is somehow possible, it'd be great! We can also do this in a follow-up PR and get this done first.

Reply:
I get your point. I started following that path, but after discussing it with @pepibumur he pointed out that working with the graph would give us more [benefits](https://github.com/tuist/tuist/pull/1683#issuecomment-678247765).

---

If I am not wrong, you can achieve this ... with:


> In my Project.swift I define that all my sources lie in MyProjectSources/**. That's why I'd expect that the documentation is only generated for them.

`tuist doc --path /path/to/Project.swift <Target>`

---

Thinking out loud (still not sure if it's possible), if there's no provided target we could:
- Load the graph
- Run swift-doc with all the targets from the graph.

---

About using the graph:
- I think it is more tuist-oriented. If we rely only on the path to the folder we want swift-doc to generate the doc, tuist becomes only a wrapper of swift-doc's binary.

Taking into account that this is my first contribution maybe we can involve more people into it.

Reply:
> If I am not wrong, you can achieve this ... with:

Well, this does not handle the case where `Project.swift` is in my root directory or does it?

> I think it is more tuist-oriented. If we rely only on the path to the folder we want swift-doc to generate the doc, tuist becomes only a wrapper of swift-doc's binary.

I don't think we should rely on the path itself, I agree - but as things stand, if I understand, we rely on the `targetPath`. So my question is if we are maybe able to leverage `graph` even *further* and extract from it the list of sources and use that list as an input to `swift-doc`

As for the case when no provided target is provided, I think your reasoning makes sense, but maybe let's first see how this feature is being used first and then we can make more informed decision how we can improve this feature 🙂

Reply:
Let me try that. I'll come back with more 😎 
Thanks for the detailed explanation

Reply:
2235cfb7

Reply:
Seems to be behaving much better now!